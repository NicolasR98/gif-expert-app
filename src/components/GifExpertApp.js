import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['Age of empires 2']);

	return (
		<>
			<h2>GifExpertApp</h2>
			<hr />
			<AddCategory setCategories={setCategories} />
			<ol>
				{categories.map((category) => (
					<GifGrid key={category} category={category} />
				))}
			</ol>
		</>
	);
};
