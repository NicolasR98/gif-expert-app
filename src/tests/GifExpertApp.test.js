import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from '../GifExpertApp';

describe('Tests on <GifExpertApp />', () => {
    const wrapper = shallow(<GifExpertApp />)
    const categories = ['doom', 'warcraft'];

    test('Should display correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Should show a list of categories', () => {
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)    
    })
    
    
})
