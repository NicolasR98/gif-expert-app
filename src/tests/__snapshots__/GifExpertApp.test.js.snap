// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> Should display correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`Tests on <GifExpertApp /> Should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="doom"
      key="doom"
    />
    <GifGrid
      category="warcraft"
      key="warcraft"
    />
  </ol>
</Fragment>
`;
